src/visualizer/raysAppFunctions/updateScenario.m

```diff
@@ -119,8 +120,15 @@ function setupTimestepInfo(app)
 
 function extractQdFilesInfo(app)
 qdFiles = dir(sprintf('%s/QdFiles',app.ns3Path));
-qdFileName = qdFiles(3).name;
-[~,~,extension]=fileparts(qdFileName);
+cd(sprintf('%s/QdFiles',app.ns3Path));
+if isfile(fullfile(cd, 'qdOutput.json'))
+   extension = '.json';
```

Comment:
it looks like there's no need for a separate case for JSON. the else case should be general enough

Reply:
I have simplified the code. check